libs/src/api/serviceProvider.js

```diff
@@ -82,35 +93,46 @@ class ServiceProvider extends Base {
       creatorNodes = creatorNodes.filter(node => !blacklist.has(node.endpoint))
     }
 
-    // Filter to healthy nodes
-    creatorNodes = (await Promise.all(
+    // Get sync status for all nodes and filter out unhealthy (unresponsive) nodes.
+    const healthyCreatorNodesSyncStatus = {}
```

Comment:
changed this bc previously it filtering out all nodes that were configured and behind. as a result, if you wanted to add a secondary and the node had previously been configured but had since droppd out of the set, you could never add it back. this code of checking syncStatus should only apply to primary selection, not secondary. so now, we do the filter for that below in primary selection.

Reply:
lgtm, i would also consider writing the below as creating two arrays first (might be a little more self-documenting)

const secondaryEligible = [] // healthy
const primaryEligible = [] // healthy and if configured is not behind

nodes.map(async node => {
  const { isBehind, isConfigured } = ....
  secondaryEligible.push(nodeEndpoint)
  if (!isConfigured || !isBehind) primaryEligible.push(nodeEndpoint)
})